falcon/util/misc.py

```diff
@@ -176,15 +180,24 @@ def http_date_to_dt(http_date: str, obs_date: bool = False) -> datetime.datetime
 
     Raises:
         ValueError: http_date doesn't match any of the available time formats
+        ValueError: http_date doesn't match allowed timezones
     """
+    date_timezone_str = http_date[-3:]
+    has_tz_identifier = not date_timezone_str.isdigit()
+    if date_timezone_str not in _ALLOWED_HTTP_TIMEZONES and has_tz_identifier:
```

Comment:
could this be lowercase/mixed case? for example is `Thu, 04 Apr 2013 00:00:00 gmt` valid?

Reply:
My interpretation of [RFC 7231, Section 7.1.1.1](https://datatracker.ietf.org/doc/html/rfc7231#section-7.1.1.1) is that the timezone field **MUST** be `GMT`. `Thu, 04 Apr 2013 00:00:00 gmt` is thus invalid.

Reply:
ok! newer rfc is https://datatracker.ietf.org/doc/html/rfc9110#section-5.6.7 btw